Hard dependency to `graphql-subscriptions`

I'm trying to use `graphql-tools` in a TypeScript project but the compilation fails with:
```
autopublish.d.ts(2,24): error TS2307: Cannot find module 'graphql-subscriptions'.
```

It looks like the module has a dependency on `graphql-subscriptions`'s `PubSub` type, but the module is not declared as a dependency.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,7 +2,7 @@
 
 ### vNEXT
 
-* ...
+* Remove dependency on `graphql-subscription` and use an interface for PubSub [PR #295](https://github.com/apollographql/graphql-tools/pull/295)
 
 ### v0.10.1
 * Update dependencies [PR #287](https://github.com/apollographql/graphql-tools/pull/287)

--- a/src/Interfaces.ts
+++ b/src/Interfaces.ts
@@ -66,3 +66,7 @@ export interface IMockOptions {
 export interface IMockServer {
     query: (query: string, vars?: { [key: string]: any }) => Promise<ExecutionResult>;
 }
+
+export interface IPubSub {
+    publish(triggerName: string, payload: any): any;
+}

--- a/src/autopublish.ts
+++ b/src/autopublish.ts
@@ -2,10 +2,10 @@ import {
     GraphQLSchema,
     GraphQLResolveInfo,
 } from 'graphql';
-import { PubSub } from 'graphql-subscriptions';
 import { chainResolvers } from './schemaGenerator';
+import { IPubSub } from './Interfaces';
 
-export function autopublishMutationResults(schema: GraphQLSchema, pubsub: PubSub) {
+export function autopublishMutationResults(schema: GraphQLSchema, pubsub: IPubSub) {
     // decorate the mutations with your thingy
     const mutationFields = schema.getMutationType().getFields();
     Object.keys(mutationFields).forEach( fieldName => {

